src/store/reducers/AnimateMap.ts

```diff
@@ -0,0 +1,135 @@
+
+// Drops the first element of a tuple.
+// see: https://dev.to/kjleitz/comment/gb5d
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type DropFirstInTuple<T extends any[]> = T extends [arg: any, ...rest: infer U]
+  ? U
+  : T;
+
+// Gets the type of the last element of a tuple.
+// see: https://dev.to/kjleitz/comment/gb5d
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type LastInTuple<T extends any[]> = T[LengthOfTuple<DropFirstInTuple<T>>];
+
+export type AnimateMapStageOptions = LastInTuple<
+  ConstructorParameters<typeof PIXI.Application>
+>;
+
+export type AnimateMapWorldBounds = { width: number; height: number };
+
+export interface AnimateMapPoint {
+  x: number; //integer
+  y: number; //integer
+}
+
+export interface ReplicatedUserData {
+  id: string;
+  videoUrlString: string;
+  avatarUrlString: string;
+  dotColor: number; //hex
+}
+
+export interface ReplicatedUser extends AnimateMapPoint {
+  data: ReplicatedUserData;
+}
+
+export interface ReplicatedVenueData {
+  id: string;
+  videoUrlString: string;
+  imageUrlString: string;
+}
+
+export interface ReplicatedVenue extends AnimateMapPoint {
+  data: ReplicatedVenueData;
+}
+
+interface AnimateMapState {
+  stageOptions: AnimateMapStageOptions;
+  worldWidth: number;
+  worldHeight: number;
+  zoom: number;
+  users: Map<string, ReplicatedUser>;
+  usersQT: QuadTree | null;
+  venues: Map<string, ReplicatedVenue>;
+  venuesQT: QuadTree | null;
+}
+
+const initialAnimateMapState: AnimateMapState = {
+  stageOptions: {
+    backgroundColor: 0x10bb99,
+    antialias: true,
+    autoDensity: true,
+    resizeTo: undefined,
+  },
+  worldWidth: 9920,
+  worldHeight: 9920,
+  zoom: 5,
+  users: new Map<string, ReplicatedUser>(),
+  usersQT: null,
+  venues: new Map<string, ReplicatedVenue>(),
+  venuesQT: null,
+};
+
+export type AnimateMapReducer = Reducer<AnimateMapState, AnimateMapActions>;
+
+function generateNewQuadtree(
+  items: Map<string, ReplicatedVenue | ReplicatedUser>,
+  quadTree: QuadTree | null,
+  state: AnimateMapState
+): QuadTree {
+  if (!quadTree) {
+    // create tree
+    quadTree = new QuadTree(
+      new Box(0, 0, state.worldWidth, state.worldHeight),
+      { maximumDepth: 100 },
+      Array.from(items).map(([key, value]) => value)
+    );
+    return quadTree;
+  } else {
+    // update tree
+    quadTree.clear(); //NOTE: can optimize if we will remove certain elements before update
+    quadTree.insert(Array.from(items).map(([key, value]) => value));
+    return quadTree;
+  }
+}
+
+export const animateMapReducer: AnimateMapReducer = (
+  state = initialAnimateMapState,
+  action: AnimateMapActions
+): AnimateMapState => {
+  switch (action.type) {
+    case AnimateMapActionTypes.UPDATE_STAGE_OPTIONS:
+      const { options } = action.payload;
+      return { ...state, stageOptions: options };
+
+    case AnimateMapActionTypes.UPDATE_ZOOM:
+      const { zoom } = action.payload;
+      return { ...state, zoom: zoom };
```

Comment:
In cases like this where the key and the value are named the same, you can use the following shorthand:
```suggestion
      return { ...state, zoom };
```